edi_storage_oca/migrations/16.0.1.0.0/pre-migrate.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+from odoo.tools.sql import create_column
+
+from odoo.addons.http_routing.models.ir_http import slugify
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if not version:
+        return
+
+    required_modules = ("storage_backend", "edi_storage_oca")
+    if not env["ir.module.module"].search(
+        [("name", "in", required_modules), ("state", "=", "installed")]
+    ):
+        return
+    if not openupgrade.column_exists(env.cr, "edi_backend", "new_storage_id"):
+        create_column(env.cr, "edi_backend", "new_storage_id", "int4")
+
+    _create_fs_storage_records(env)
+
+    if openupgrade.column_exists(env.cr, "edi_backend", "new_storage_id"):
+        _column_renames = {
+            "edi_backend": [
+                ("storage_id", "old_storage_id"),
+                ("new_storage_id", "storage_id"),
+            ],
+        }
+        openupgrade.rename_columns(env.cr, _column_renames)
+
+
+def _create_fs_storage_records(env):
+    # Create a fs.storage record for each backend.storage
+    storage_backend_records = env["storage.backend"].search([])
```

Comment:
I think we should only transform the backends that come from edi.backends. Probably we need to add some SQL here, but it is the cleaner option IMO